Re-wrote snapchat

waveform80 opened the pull request:
Not suggesting this is actually merged - I have done away with the simple/advanced examples because I haven't got time to re-write them just now, but you might find some ideas in this version that would be useful:
- Simplified the overlay setup and updates
- Cached overlay images so it's a bit faster at switching
- Uses 4:3 resolution to avoid offset problem
- Simplified the alpha compositing (just use paste(..., mask=...))
- Cropped all the overlays down (and made the sunglasses
  semi-transparent :)
- Added a couple more buttons to move the overlay up and down

waveform80 committed:
Re-wrote snapchat

* Simplified the overlay setup and updates
* Cached overlay images so it's a bit faster at switching
* Uses 4:3 resolution to avoid offset problem
* Simplified the alpha compositing (just use paste(..., mask=...))
* Cropped all the overlays down (and made the sunglasses
  semi-transparent :)
* Added a couple more buttons to move the overlay up and down

waveform80 commented:
Oh, and there's some silly bits: the MCP3008 import can be removed (I was experimenting with using a pot to set the overlay position); and I've probably used the wrong GPIOs for the buttons (I just used whatever was most convenient on a breakout).

@bennuttall ^^

bennuttall commented:
Cool thanks @waveform80 - will be handy for next time we want to run this, or if we turn it into a resource or something.

waveform80 committed:
Ooops ... missed a couple of bits

The (as yet unused) global font wasn't declared global; also tidied up a
few bits (no need to calculate a full bounding box for the overlay
paste) and removed MCP3008
